relayer/chains/wasm/provider.go

```diff
@@ -129,19 +129,21 @@ func (p *Provider) Listener(ctx context.Context, lastSavedHeight uint64, blockIn
 		case <-subscribeStarter.C:
 			subscribeStarter.Stop()
 			for _, event := range p.contracts {
-				go p.SubscribeMessageEvents(ctx, blockInfoChan, &types.SubscribeOpts{
-					Address: event.Address,
-					Method:  event.GetWasmMsgType(),
-					Height:  latestHeight,
-				}, resetFunc)
+				for msgType := range event.SigType {
+					go p.SubscribeMessageEvents(ctx, blockInfoChan, &types.SubscribeOpts{
+						Address: event.Address,
+						Method:  msgType,
+						Height:  latestHeight,
+					}, resetFunc)
+				}
 			}
 		case <-pollHeightTicker.C:
 			pollHeightTicker.Stop()
 			startHeight = p.GetLastSavedHeight()
 			latestHeight, err = p.QueryLatestHeight(ctx)
 			if err != nil {
 				p.logger.Error("failed to get latest block height", zap.Error(err))
-				continue
+				pollHeightTicker.Reset(time.Second * 3)
```

Comment:
what change would this logic have since we previously had pollHeightTicker tied to sibscribeStarted ticket but now they can be independent ? can you explain on the reason for this change which scenario would be fixed by this change ?